internal/bundle/utils.go

```diff
@@ -116,25 +179,59 @@ func LoadBundleFromDisk(path, name string, bvc *bundle.VerificationConfig) (*bun
 	}
 }
 
-// SaveBundleToDisk saves the given raw bytes representing the bundle's content to disk
-func SaveBundleToDisk(path string, raw io.Reader) (string, error) {
+// SaveBundleToDisk persists a bundle to disk. Bundles are wrapped in a 'bundlePackage' in order
+// to support additional metadata (e.g. etag) that is not part of the original bundle.
+func SaveBundleToDisk(path string, rawBundle io.Reader, opts *SaveOptions) error {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
 		err = os.MkdirAll(path, os.ModePerm)
 		if err != nil {
-			return "", err
+			return err
 		}
 	}
 
-	if raw == nil {
-		return "", fmt.Errorf("no raw bundle bytes to persist to disk")
+	// supplying no bundle data is an error case
+	if rawBundle == nil {
+		return fmt.Errorf("no raw bundle bytes to persist to disk")
+	}
+
+	// create a temporary file to write the bundlepackage to
+	destBundlePackage, err := os.CreateTemp(path, ".bundlePackage.tar.gz.*.tmp")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary bundle package file: %w", err)
+	}
+	defer destBundlePackage.Close()
+
+	rawBundleBytes, err := io.ReadAll(rawBundle)
```

Comment:
I feel like this is marginally worse than what we were able to do before since it requires copying the bundle bytes to memory before marshalling as JSON. I'm unsure if there's a way to io.Copy the bytes of the raw bundle straight into the file in some way.